core/src/main/java/com/radixdlt/api/system/routes/HealthHandler.java

```diff
@@ -122,24 +130,85 @@ public HealthResponse handleRequest() {
                             p,
                             readinessSignalStatuses.getOrDefault(
                                 p.protocolUpdateTrigger().nextProtocolVersion(),
-                                NO_SIGNAL_REQUIRED)))
+                                NO_SIGNAL_REQUIRED),
+                            ledgerSummary))
                 .toList());
   }
 
   private com.radixdlt.api.system.generated.models.PendingProtocolUpdate pendingProtocolUpdate(
       ProtocolState.PendingProtocolUpdate pendingProtocolUpdate,
-      PendingProtocolUpdate.ReadinessSignalStatusEnum readinessSignalStatus) {
-    final var state =
-        switch (pendingProtocolUpdate.state()) {
-          case ProtocolState.PendingProtocolUpdateState.Empty
-          empty -> new EmptyPendingProtocolUpdateState().type(PendingProtocolUpdateStateType.EMPTY);
-          case ProtocolState.PendingProtocolUpdateState.ForSignalledReadinessSupportCondition
-          forSignalledReadinessSupportCondition -> new SignalledReadinessPendingProtocolUpdateState()
-              .thresholdsState(
-                  forSignalledReadinessSupportCondition.thresholdsState().stream()
-                      .map(
-                          thresholdState ->
-                              new SignalledReadinessPendingProtocolUpdateStateAllOfThresholdsState()
+      PendingProtocolUpdate.ReadinessSignalStatusEnum readinessSignalStatus,
+      LedgerSummary ledgerSummary) {
+    final var currentEpoch = ledgerSummary.currentEpoch();
+    final var epochStartTimeCalculator =
+        new EpochStartTimeCalculator(
+            ledgerSummary.consensusManagerConfigEpochTargetDurationMs(),
+            currentEpoch,
+            ledgerSummary.consenusManagerStateEpochEffectiveStartMs());
+
+    return switch (pendingProtocolUpdate.protocolUpdateTrigger().enactmentCondition()) {
+      case EnactAtStartOfEpochIfValidatorsReady condition -> {
```

Comment:
To make it easier to verify line-by-line, I'd rather this were called `atStartOfEpochCondition`